core/templates/dev/head/pages/Base.ts

```diff
@@ -55,6 +58,11 @@ oppia.controller('Base', [
       $scope.currentLang = response.language;
     });
 
+    if (UrlService.getPathname() !== '/signup') {
+      $http.get('/csrf').then(function(response) {
```

Comment:
Also I think it might be better to move this into a function in CsrfTokenService called CsrfTokenService.populateNewCsrfToken() -- better to get most of this functionality into services, so that it can be easily tested. Have a check to ensure that this function is called only once per page; throw an error if it's called a second time.

Reply:
Putting this function in the service creates a circular dependency. Because the token fetched is used in the `$httpProvider` interceptor.

Reply:
I see, OK. Please add a line comment to explain: "This call is not part of CsrfTokenService because ..."